### PR TITLE
Ubuntu/jammy: fix dailies

### PR DESCRIPTION
**Pull request by TheRealFalcon**
@holmanb , do you mind reviewing this one? There were two conflicts based on some of your recent changes and I want to make sure I didn't mess them up.

I made an accidental bad push to upstream jammy, so it is currently at tip of main even though it was originally further back.

### CODE REVIEW

File: debian/patches/revert-551f560d-cloud-config-after-snap-seeding.patch [hunk 1]
```diff
@@ -147,22 +147,21 @@ Last-Update: 2024-02-14
...
          for backend, cmd, package in BACKEND_DEF:
-@@ -88,23 +84,21 @@ class TestLxd(t_help.CiTestCase):
-             if backend == "lvm":
-                 self.assertEqual(
-                     [
+@@ -85,22 +81,14 @@ class TestLxd(t_help.CiTestCase):
+                 subp.call_args_list,
+             )
+
+-            if backend == "lvm":
```
Comment (TheRealFalcon) [inline debian/patches/revert-551f560d-cloud-config-after-snap-seeding.patch:154]: Is this right?

File: debian/patches/no-single-process.patch [hunk 2]
```diff
@@ -40,9 +40,9 @@ Last-Update: 2024-08-02
  RemainAfterExit=yes
  TimeoutSec=0
 +KillMode=process
- {% if variant in ["almalinux", "cloudlinux", "rhel"] %}
- # Restart NetworkManager if it is present and running.
- ExecStartPost=/bin/sh -c 'u=NetworkManager.service; \
+ TasksMax=infinity
```
Comment (TheRealFalcon) [inline debian/patches/no-single-process.patch:43]: Is this right?

**Comment by holmanb** [comment]
It looks like I broke the GH Actions quilt job, [here is a fix](https://github.com/canonical/cloud-init/pull/5713).

That test passes manually.

File: debian/patches/revert-551f560d-cloud-config-after-snap-seeding.patch (hunk 1 above)
Comment (holmanb) [reply debian/patches/revert-551f560d-cloud-config-after-snap-seeding.patch:154]: This is... confusing. The original test checks to make sure that the semaphore exists when waiting for snap. But on jammy, I'm not sure why we are still testing for this call. Am I missing something?

Comment (TheRealFalcon) [reply debian/patches/revert-551f560d-cloud-config-after-snap-seeding.patch:154]: we were testing the `dm-thin-pool.ko` that was just removed, so I removed that as well.

Comment (holmanb) [reply debian/patches/revert-551f560d-cloud-config-after-snap-seeding.patch:154]: > we were testing the `dm-thin-pool.ko` that was just removed, so I removed that as well.

I know, I'm commenting on the remaining test. I don't think that it tests anything valuable, but I guess it doesn't hurt anything either.

**Review by holmanb — APPROVED** [review]
LGTM

**Pull request merged**